pkg/scheduler/tests/scheduler_recovery_test.go

```diff
@@ -667,4 +667,65 @@ partitions:
 		GetSchedulingApplication("app-1", "[rm:123]default")
 	assert.Equal(t, app01.ApplicationInfo.ApplicationId, "app-1")
 	assert.Equal(t, app01.ApplicationInfo.QueueName, "root.a")
+}
+
+// test scheduler recovery that only registers nodes and apps
+func TestAppRecoveryAlone(t *testing.T) {
+	serviceContext := entrypoint.StartAllServicesWithManualScheduler()
+	proxy := serviceContext.RMProxy
+
+	// Register RM
+	configData := `
+partitions:
+  -
+    name: default
+    queues:
+      - name: root
+        queues:
+          - name: a
+            resources:
+              guaranteed:
+                memory: 100
+                vcore: 10
+              max:
+                memory: 150
+                vcore: 20
+`
+	configs.MockSchedulerConfigByData([]byte(configData))
+	mockRM := NewMockRMCallbackHandler(t)
+
+	_, err := proxy.RegisterResourceManager(
+		&si.RegisterResourceManagerRequest{
+			RmId:        "rm:123",
+			PolicyGroup: "policygroup",
+			Version:     "0.0.2",
+		}, mockRM)
+
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	// Register apps alone
+	err = proxy.Update(&si.UpdateRequest{
+		NewApplications: []*si.AddApplicationRequest{
+			{
+				ApplicationId: "app-1",
+				QueueName:     "root.a",
+				PartitionName: "",
```

Comment:
Do we need to specify resource request also here, such as memory or cpu ?

Reply:
Nope, this is not needed for registering an app per scheduler-interface. See more at https://github.com/cloudera/yunikorn-scheduler-interface/blob/d992f6689cd7a4b81f750f7b1a57f6cc7c677857/si.proto#L239-L243